feat: add support for 5.1

garlicNova opened the pull request:
<!-- markdownlint-disable first-line-h1 no-inline-html -->

<!--
    In order to have the best experience with our community, we recommend that you read the code of conduct and contributing guidelines before submitting a pull request.
    
    By submitting this pull request, you confirm that you have read, understood, and agreed to the project's code of conduct and contributing guidelines.

    Please use conventional commits to format the title of the pull request and the commit messages.
    For more information, please refer to https://www.conventionalcommits.org.
-->

### Summary

- Added 5.1.0.0 to  `Get-PasswordPolicyDefault` and `Get-PasswordPolicyConfig` to support VMware Cloud Foundation 5.1.
- Updated `Update-SddcManagerPasswordComplexity`, `Update-SddcManagerAccountLockout`, and `Update-vCenterAccountLockout` cmdlet to support VMware Cloud Foundation 5.1 by addressing changes in Photon OS 4 for vCenter Server 8.0.2 and SDDC Manager 5.1. 

### Type

<!--
    Please check the one(s) that applies to this pull request using "x".
-->

- [ ] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

<!--
    Please check the one that applies to this pull request using "x".
    If this pull request contains a breaking change, please describe the impact and mitigation path.
-->

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

<!--
    Please check the one(s) that applies to this pull request using "x".
    For bug fixes and enhancements/features, please ensure that tests and documentation have been completed and provide details.
-->

- [x] Tests have been completed.
- [ ] Documentation has been added or updated.

<!--
    Please describe the tests that have been completed and/or the documentation that has been added/updated.
-->

### Issue References

<!--
    Is this related to any GitHub issue(s)? If so, please provide the issue number(s) that are closed or resolved by this pull request.

    For bug fixes and enhancements/features, please ensure that a GitHub issue has been created and provide the issue number(s) here.

    Please use the 'Closes' keyword followed by the a hash and issue number.
    This will link the pull request to the issue(s) and automatically close them when the pull request is merged.

    Example:

    Closes #000
    Closes #001
-->
Closes #129 

### Additional Information

<!--
    Please provide any additional information that may be helpful.
-->

Commit:
feat: add support VCF 5.1 accountLockout

fix support for VCF 5.1 with vcenter account Lockout

update-vcenterAccountLockout

Signed-off-by: Kevin Teng <kteng@vmware.com>

Commit:
feat: add support VCF 5.1 accountLockout

fix support for VCF 5.1 with vcenter account Lockout

update-vcenterAccountLockout

Signed-off-by: Kevin Teng <kteng@vmware.com>

tenthirtyam commented:
I've added one comment.

Please ensure the module version/build is updated along with the `CHANGELOG.md` too.

Example:

```powershell
1.6.0.1000
```

Example:

```markdown
## 1.6.0

> Unreleased
```

tenthirtyam commented:
Are the same changes required for SDDC Manager?

Commit:
feat: add support for VCF 5.1

updated `Update-SddcManagerPasswordComplexity`, `Update-SddcManagerAccountLockout`, `Update-vCenterAccountLockout` cmdlet to address changes in Photon OS 4 for vCenter 8.0.2 and SDDC Manager 5.1

Signed-off-by: Kevin Teng <kteng@vmware.com>

tenthirtyam commented on VMware.CloudFoundation.PasswordManagement.psm1:
```diff
@@ -2401,8 +2401,28 @@ Function Update-SddcManagerPasswordComplexity {
-                            if ($chkExistingConfig) {
-                                Set-LocalPasswordComplexity -vmName ($server.Split("."))[-0] -guestUser root -guestPassword $rootPass -minLength $minLength -uppercase $minUppercase -lowercase $minLowercase -numerical $minNumerical -special $minSpecial -unique $minUnique -class $minClass -sequence $maxSequence -history $history -retry $maxRetry | Out-Null
+                            $sddcManagerVersion = Get-VCFManager -version
+                            if (($sddcManagerVersion.split(".")[0] -eq 5 -and $sddcManagerVersion.split(".")[1] -ge 1) -and ($existingConfiguration.'Min Length' -eq $null -or $existingConfiguration.'Min Lowercase' -eq $null -or $existingConfiguration.'Min Uppercase' -eq $null -or $existingConfiguration.'Min Numerical' -eq $null -or $existingConfiguration.'Min Special' -eq $null -or $existingConfi ...
```
Shouldn't this be `-ge`?

Otherwise, this must be revisited for any release after 5.1.

tenthirtyam commented on VMware.CloudFoundation.PasswordManagement.psm1:
```diff
@@ -2488,10 +2508,22 @@ Function Update-SddcManagerAccountLockout {
-                                $updatedConfiguration = Get-LocalAccountLockout -vmName ($server.Split("."))[-0] -guestUser root -guestPassword $rootPass -product sddcManager
+                            $existingConfiguration = Get-LocalAccountLockout -vmName ($server.Split("."))[0] -guestUser root -guestPassword $rootPass -product sddcManager
+                            $sddcManagerVersion = Get-VCFManager -version
+                            if(($sddcManagerVersion.split(".")[0] -eq 5 -and $sddcManagerVersion.split(".")[1] -eq 1) -and ($existingConfiguration.'Max Failures' -eq $null -or $existingConfiguration.'Unlock Interval (sec)' -eq $null -or $existingConfiguration.'Root Unlock Interval (sec)' -eq $null)){
```
Shouldn't this be `-ge`?

Otherwise, this must be revisited for any release after 5.1.

garlicNova replied:
addressed in 1e1b3c355f69df44732a9b14237d38310a446427

tenthirtyam reviewed (changes requested):
@garlicNova - I've rebased this to resolve the signing issue and updated the `CHANGELOG.md`. 

There are three comments to be addressed and we'll be good to go.

Commit:
fix: address all current comments.

fix: address all current comments.

Signed-off-by: Kevin Teng <kteng@vmware.com>

tenthirtyam commented:
@garlicNova - I've rebased this to resolve the signing issue.

tenthirtyam reviewed (approved):
LGTM!
